### PR TITLE
Improve startup plugin collection.

### PR DESCRIPTION
**Pull request by bsneed**
- Moved required per-platform plugins to VendorSystem.
- Refactored startup sequence to fetch from VendorSystem and then apply lifecycle stuff.

### CODE REVIEW

File: Sources/Segment/Startup.swift [hunk 1]
```diff
@@ -40,20 +40,24 @@ extension Analytics: Subscriber {
...
         // this must come first.
         plugins.append(Context())
+        
+        plugins += VendorSystem.current.requiredPlugins
 
         // setup lifecycle if desired
         if configuration.values.trackApplicationLifecycleEvents {
             #if os(iOS) || os(tvOS)
-            plugins += [iOSLifecycleMonitor(), iOSLifecycleEvents(), DeviceToken()]
+            plugins.append(iOSLifecycleEvents())
```
Comment (prayansh) [inline Sources/Segment/Startup.swift:49]: would it make more sense to provide the lifecycleEvents plugin via the vendor system as well?

Comment (bsneed) [reply Sources/Segment/Startup.swift:49]: I think that would add clutter in the long term if/when we have more things that fall into that bucket.

**Review by prayansh — APPROVED** [review]
LGTM, just a suggestion on providing the lifecycleEvent plugin via vendor system, to potentially remove macro code

**Review by migs647 — APPROVED** [review]
LGTM

**Pull request merged**